apps/web/src/contexts/globalLayout/GlobalSidebar/AnimatedSidebarDrawer.tsx

```diff
@@ -0,0 +1,157 @@
+import { motion } from 'framer-motion';
+import { MouseEvent, ReactElement, useCallback, useMemo, useRef } from 'react';
+import styled from 'styled-components';
+
+import breakpoints from '~/components/core/breakpoints';
+import { Button } from '~/components/core/Button/Button';
+import colors from '~/components/core/colors';
+import IconContainer from '~/components/core/IconContainer';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { TitleDiatypeL } from '~/components/core/Text/Text';
+import {
+  ANIMATED_COMPONENT_TRANSITION_S,
+  ANIMATED_COMPONENT_TRANSLATION_PIXELS_SMALL,
+  rawTransitions,
+} from '~/components/core/transitions';
+import useDetectOutsideClick from '~/hooks/useDetectOutsideClick';
+import useKeyDown from '~/hooks/useKeyDown';
+import { useIsMobileWindowWidth } from '~/hooks/useWindowSize';
+import CloseIcon from '~/icons/CloseIcon';
+
+import { useDrawerActions } from './SidebarDrawerContext';
+
+type Props = {
+  content: ReactElement;
+  headerText?: string;
+  showDoneFooter?: boolean;
+};
+
+export default function AnimatedSidebarDrawer({
+  content,
+  headerText,
+  showDoneFooter = false,
+}: Props) {
+  const drawerRef = useRef(null);
+
+  const { hideDrawer } = useDrawerActions();
+
+  useDetectOutsideClick(drawerRef, hideDrawer);
+
+  useKeyDown('Escape', hideDrawer);
+
+  // Prevent useDetectOutsideClick from triggering if the user clicks inside the drawer.
+  // This is needed for cases when the user clicks on elements that dissapear upon clicking, like the "See More" button in Notifications.
+  const handleDrawerClick = useCallback((e: MouseEvent) => {
+    e.stopPropagation();
+  }, []);
+
+  const handleCloseDrawerClick = useCallback(() => {
+    hideDrawer();
+  }, [hideDrawer]);
+
+  const handleDoneClick = useCallback(() => {
+    hideDrawer();
+  }, [hideDrawer]);
+
+  const isMobile = useIsMobileWindowWidth();
+
+  const motionSettings = useMemo(() => {
```

Comment:
nit: declaring a return type on the function inside the useMemo would get us better autocomplete inside the function